org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/editor/quickfix/XtextGrammarQuickfixProvider.java

```diff
@@ -474,4 +479,40 @@ public Integer exec(XtextResource state) throws Exception {
 			}
 		});
 	}
+
+	@Fix(RuleWithoutInstantiationInspector.ISSUE_CODE)
+	public void addAction(final Issue issue, IssueResolutionAcceptor acceptor) {
+		acceptor.accept(issue, "Add actions to ensure object creation", "Inserts the missing actions to ensure object creation.", null,
+				new ISemanticModification() {
+					@Override
+					public void apply(EObject element, IModificationContext context) throws Exception {
+						ParserRule rule = (ParserRule) element;
+						MultiTextEdit textEdit = new MultiTextEdit();
+						applyToNode(rule.getAlternatives(), "{" + rule.getName() + "} ", textEdit);
```

Comment:
The action must produce an instance of the type that is returned by the rule. It may need to be qualified with the metamodels alias. The code as is is likely to produce invalid grammars. Another thing is, that the return type may be an abstract type.

Reply:
I have introduceed code to handle "returns" even for static models. Writing tests for the cases where the model is not generated will be quite a lot of effort. There are no tests yet at all for static models which means to setup (and probably refactor) testing infrastructure.

Would you be able to guide me to examples about cases where the rule needs to be qualified or the returned type is abstract? I was not able to come to such examples by my own. Maybe I miss something important here.

Reply:
My latest comit: https://github.com/ArneDeutsch/xtext-eclipse/commit/a25f740de01b2aea7416b77c9e2cdb14389495ab

Reply:
A good starting point would be to use a `generate` clause with an alias, e.g. `generate myDsl "http://www.xtext.org/example/mydsl/MyDsl" as alias` and `Model returns alias::Model: ...` respectively.

Reply:
Added code and test to handle alias as well.